File: app/src/main/java/com/lcarino/bucketlist/mvp/MvpPresenter.java
package com.lcarino.bucketlist.mvp;

import android.support.annotation.UiThread;

/**
 * The base interface for mvp presenters.
 *
 * @author Luis Carino
 */

public interface MvpPresenter<V extends MvpView> {

    /**
     * Set o attach the view to its getListPresenter.
     * @param view
     */
    @UiThread
    void attachView(V view);

    /**
     * Will be called if the view has been destroyed. Typically this method will be invoked from
     * <code>Activity.detachView()</code> or <code>Fragment.onDestroyView()</code>
     */
    @UiThread
    void detachView(boolean retainState);

}

File: app/src/main/java/com/lcarino/bucketlist/mvp/kotlin/MvpPresenter.kt
package com.lcarino.bucketlist.mvp.kotlin

import android.os.Bundle
import android.support.annotation.Nullable

/**
 * Base interface for MVP presenters.
 * Created by luiscarino on 3/8/17.
 */
interface MvpPresenter<V : View> {
    fun onCreateView(view: V, @Nullable bundle: Bundle)
    fun onResume()
    fun onSaveInstanceState()
    fun onPause()
    fun onDetachView()
}